scipy/stats/_discrepancy.pyx

```diff
@@ -0,0 +1,502 @@
+# cython: language_level=3
+# cython: boundscheck=False
+# cython: wraparound=False
+# cython: cdivision=True
+
+# distutils: language = c++
+
+import numpy as np
+cimport numpy as np
+from libc.math cimport fabs, sqrt, pow
+
+np.import_array()
+
+cdef extern from "<thread>" namespace "std" nogil:
+    cdef cppclass thread:
+        thread()
+        void thread[A, B, C, D, E, F](A, B, C, D, E, F)
+        void join()
+
+cdef extern from "<mutex>" namespace "std" nogil:
+    cdef cppclass mutex:
+        void lock()
+        void unlock()
+
+cdef extern from "<functional>" namespace "std" nogil:
+    cdef cppclass reference_wrapper[T]:
+        pass
+    cdef reference_wrapper[T] ref[T](T&)
+
+from libcpp.vector cimport vector
+
+
+cdef mutex threaded_sum_mutex
+
+
+def discrepancy(sample, bint iterative=False, method='CD', workers=1):
+    """Discrepancy of a given sample.
+
+    Parameters
+    ----------
+    sample : array_like (n, d)
+        The sample to compute the discrepancy from.
+    iterative : bool, optional
+        Must be False if not using it for updating the discrepancy.
+        Default is False. Refer to the notes for more details.
+    method : str, optional
+        Type of discrepancy, can be ``CD``, ``WD``, ``MD`` or ``L2-star``.
+        Refer to the notes for more details. Default is ``CD``.
+    workers : int, optional
+        If `workers` is greater than 1, the population is subdivided into
+        `workers` sections and evaluated in parallel (uses
+        c++ threads).
```

Comment:
I think a description (and behaviour) like http://scipy.github.io/devdocs/generated/scipy.spatial.cKDTree.query.html?highlight=workersis preferred. In particular, allowing `-1`.

Reply:
Done in e6864f285b5c8c7d0734483d019664ed3460f249. Also add tests for parallelisms